forte/data/converter/converter.py

```diff
@@ -0,0 +1,343 @@
+        masks_list: List[List[Any]] = []
+        for i in range(features[0].dim):
+            curr_dim_masks = []
+            for mask in masks_per_example_list:
+                curr_dim_masks.append(mask[i])
+            masks_list.append(curr_dim_masks)
+
+        # Convert to target type
+        if not self.to_numpy and not self.to_torch:
+            return data_list, masks_list
+
+        # Note: to_torch == True overwrite to_numpy option
+        if self.to_torch:
+            data_tensor: torch.Tensor = \
+                self._to_tensor_type(data_list, dtype)
+            masks_tensor_list: List[torch.Tensor] = []
+            for batch_masks_dim_i in masks_list:
+                masks_tensor_list.append(
+                    self._to_tensor_type(batch_masks_dim_i, np.bool))
+
+            return data_tensor, masks_tensor_list
+
+        if self.to_numpy:
+            data_np: np.ndarray = \
+                self.to_numpy_type(data_list, dtype)
+            masks_np_list: List[np.ndarray] = []
+            for batch_masks_dim_i in masks_list:
+                masks_np_list.append(
+                    self.to_numpy_type(batch_masks_dim_i, np.bool))
+
+            return data_np, masks_np_list
+
+        # Control should not reach here
+        raise RuntimeError("Invalid converter internal state")
+
+    @staticmethod
+    def _padding(features: List[Feature]):
+        # BFS to pad each dimension
+        queue: List[Feature] = []
+        curr_max_len: int = -1
+        dtype: Optional[torch.dtype] = None
+
+        for feature in features:
+            if not dtype:
+                dtype = feature.dtype
+            else:
+                if dtype != feature.dtype:
+                    raise ValidationError(
+                        "The dtype should be same within a batch of Features")
+            if not feature.need_pad:
+                raise ValidationError(
+                    "Inconsistent need pad flag for a batch of Features")
+            queue.append(feature)
+            curr_max_len = max(curr_max_len, len(feature))
+
+        while len(queue) > 0:
+            size: int = len(queue)
+            next_max_len = -1
+            while size > 0:
+                feature = queue.pop(0)
+                feature.pad(curr_max_len)
+
+                if not feature.leaf_feature:
+                    for sub_feature in feature.sub_features:
+                        next_max_len = max(next_max_len,
+                                           len(sub_feature))
+                        queue.append(sub_feature)
+
+                size -= 1
+
+            curr_max_len = next_max_len
+
+        return dtype
+
+    @staticmethod
+    def to_numpy_type(data: List[Any], dtype) -> np.ndarray:
+        return np.array(data, dtype=dtype)
+
+    @staticmethod
+    def _to_tensor_type(data: List[Any], dtype) -> torch.Tensor:
+        return torch.tensor(data, dtype=dtype)
```

Comment:
Why one of these function is protected (with `_`), and one is not?

Reply:
Changed `to_numpy_type` to `_numpy_type`